server/server.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/allegro/bigcache"
+)
+
+const (
+	apiVersion  = "v1"
+	apiBasePath = "/api/" + apiVersion + "/"
+
+	// path to cache.
+	cachePath = apiBasePath + "cache/"
+)
+
+var (
+	lifeWindow int64
+	hardMax    int
+	verbose    bool
+	logfile    string
+	port       int
+	cache      *bigcache.BigCache
+)
+
+func init() {
+	flag.BoolVar(&verbose, "v", false, "Verbose logging.")
+	flag.Int64Var(&lifeWindow, "lifetime", 10, "Lifetime, in minutes, of each cache object.")
+	flag.IntVar(&hardMax, "max", 8192, "Maximum amount of objects in the cache.")
+	flag.IntVar(&port, "port", 9090, "The port to listen on.")
+	flag.StringVar(&logfile, "logfile", "", "Location of the logfile.")
+}
+
+func main() {
+	flag.Parse()
+
+	var logger *log.Logger
+
+	if logfile == "" {
+		logger = log.New(os.Stdout, "", log.LstdFlags)
+	} else {
+		f, err := os.OpenFile(logfile, os.O_APPEND|os.O_WRONLY, 0600)
+		if err != nil {
+			panic(err)
+		}
+		logger = log.New(f, "", log.LstdFlags)
+	}
+
+	config := bigcache.Config{
+		Shards:             1024,
+		LifeWindow:         time.Duration(lifeWindow) * time.Minute,
+		MaxEntriesInWindow: 1000 * 10 * 60,
+		MaxEntrySize:       500,
```

Comment:
How about making `Shards`, `MaxEntriesInWindow` and `MaxEntriesInWindow` configurable too?

Reply:
Sure, I don't see why not.